snippets/src/issue-a-credential.ts

```diff
@@ -0,0 +1,203 @@
+      },
+    ],
+    autoAcceptCredentials: AutoAcceptCredential.Always,
+    autoAcceptConnections: true,
+  }
+
+  // A new instance of an agent is created here
+  const agent = new Agent(config, agentDependencies)
+
+  // Register a simple `WebSocket` outbound transport
+  agent.registerOutboundTransport(new WsOutboundTransport())
+
+  // Register a simple `Http` outbound transport
+  agent.registerOutboundTransport(new HttpOutboundTransport())
+
+  // Initialize the agent
+  await agent.initialize()
+
+  return agent
+}
+// end-section-1
+
+// start-section-2
+const initializeIssuerAgent = async () => {
+  const genesisTransactionsBCovrinTestNet = await getGenesisTransaction("http://test.bcovrin.vonx.io/genesis")
+  // Simple agent configuration. This sets some basic fields like the wallet
+  // configuration and the label.
+  const config: InitConfig = {
+    label: "demo-agent-issuer",
+    walletConfig: {
+      id: "demo-agent-issuer",
+      key: "demoagentissuer00000000000000000",
+    },
+    publicDidSeed: "demoissuerdidseed000000000000000",
+    indyLedgers: [
+      {
+        id: "bcovrin-test-net",
+        isProduction: false,
+        genesisTransactions: genesisTransactionsBCovrinTestNet,
+      },
+    ],
+    autoAcceptCredentials: AutoAcceptCredential.Always,
+    autoAcceptConnections: true,
+    endpoints: ["http://localhost:3001"],
+  }
+
+  // A new instance of an agent is created here
+  const agent = new Agent(config, agentDependencies)
+
+  // Register a simple `WebSocket` outbound transport
+  agent.registerOutboundTransport(new WsOutboundTransport())
+
+  // Register a simple `Http` outbound transport
+  agent.registerOutboundTransport(new HttpOutboundTransport())
+
+  // Register a simple `Http` inbound transport
+  agent.registerInboundTransport(new HttpInboundTransport({ port: 3001 }))
+
+  // Initialize the agent
+  await agent.initialize()
+
+  return agent
+}
+// end-section-2
+
+// start-section-3
+const registerSchema = async (issuer: Agent) =>
+  issuer.ledger.registerSchema({ attributes: ["name", "age"], name: "Identity", version: "6.0" })
+// end-section-3
+
+// start-section-4
+const registerCredentialDefinition = async (issuer: Agent, schema: Schema) =>
+  issuer.ledger.registerCredentialDefinition({ schema, supportRevocation: false, tag: "default" })
+// end-section-4
+
+// start-section-5
+const setupCredentialListener = (holder: Agent) => {
+  holder.events.on<CredentialStateChangedEvent>(CredentialEventTypes.CredentialStateChanged, async ({ payload }) => {
+    switch (payload.credentialRecord.state) {
+      case CredentialState.OfferReceived:
+        console.log("received a credential")
+        // custom logic here
+        await holder.credentials.acceptOffer({ credentialRecordId: payload.credentialRecord.id })
+    }
+  })
+}
+// end-section-5
+
+// start-section-6
+const issueCredential = async (issuer: Agent, credentialDefinitionId: string, connectionId: string) =>
```

Comment:
Why is everything in methods? The tutorial didn't have this right?

Reply:
Everything is a method to make way cleaner snippet sections. Indentation got weird and this also show what is required for that small part of the snippet to work.